commons/commons-bio/src/main/kotlin/ebi/ac/uk/io/sources/FilesSource.kt

```diff
@@ -5,9 +5,13 @@ import ebi.ac.uk.io.ext.size
 import java.io.File
 
 interface FilesSource {
+    val rootPath: String?
```

Comment:
this looks incorrect for me, root path is not related at all with concept of source (a generic interface that allow to obtain a file). This is coupling this class with an special feature of submitter that allow use to locate files in a folder which in my opinion is not related at all. 

You can notice this is not a good modeling option as for example ComposedFileSource now have a root path which does not make sense given compose only is a file source compoused of others files not relation at all.

Reply:
I think what we need to do is change interface so getFile return nullable file  `BioFile?` and callers trigger exception with proper message which actually make sense as only caller knows what error to trigger and not the individual source which is totally normal they fail to find a fail.